File: Case2/Filters2/h.py
import numpy as np

def h(x):
    if np.size(x.shape) == 1:
        z = np.zeros((2))
        z[0] = x[0] ** 2
        z[1] = x[1]
    else:
        N = x.shape[1]
        z = np.zeros((2, N))
        for i in range(N):
            z[0, i] = x[0, i] ** 2
            z[1, i] = x[1, i]
    return z

File: Data/h.m

function z = h(x)
N = size(x,2);
z = zeros(2,N);
for i = 1:N
    z(1,i) = x(1, i)^2;
    z(2,i) = x(2, i);
end
end

% function z = h(x)
% z = x;
% end

